### PR TITLE
Refactor calculator to make it more abstract.

### PR DESCRIPTION
**Pull request by Spaceenter**
Only keep AST-related operations in calculator.go, and move all derived-series-related functions to derived_series.go. Now the code in calculator.go could be reused for obs collection calculations (entity->val).

### CODE REVIEW

File: internal/server/v1/observations/calculator.go [hunk 1]
```diff
@@ -21,39 +21,31 @@ import (
...
-	return strings.Join(dates, "")
-}
-
-type varInfo struct {
-	statVar          string
-	statMetadata     *pb.StatMetadata
-	seriesCandidates []*calculatorSeries
-	chosenSeries     *calculatorSeries
+// The info of a node in the AST tree.
+type nodeInfo struct {
```
Comment (shifucun) [inline internal/server/v1/observations/calculator.go:38]: This is more like nodeData?

File: internal/server/v1/observations/calculator.go [hunk 2]
```diff
@@ -21,39 +21,31 @@ import (
...
+	statVar        string
+	statMetadata   *pb.StatMetadata
+	itemCandidates []calcItem
```
Comment (shifucun) [inline internal/server/v1/observations/calculator.go:41]: candidateItems

**Review by shifucun — APPROVED** [review]
Nice! Just some naming comments. Look forward to see how is this generalized to obsCollections

File: internal/server/v1/observations/calculator.go [hunk 3]
```diff
@@ -21,39 +21,31 @@ import (
...
 
 	pb "github.com/datacommonsorg/mixer/internal/proto"
 	"github.com/datacommonsorg/mixer/internal/util"
 )
 
-type calculatorSeries struct {
-	statMetadata *pb.StatMetadata
-	// Sorted by date.
-	points []*pb.PointStat
+// An calculable item that belong to a node in the AST tree.
```
[the remark this answers is not in the document]
Comment (Spaceenter) [reply internal/server/v1/observations/calculator.go:31]: done

File: internal/server/v1/observations/calculator.go (hunk 1 above)
Comment (Spaceenter) [reply internal/server/v1/observations/calculator.go:38]: done

File: internal/server/v1/observations/calculator.go (hunk 2 above)
Comment (Spaceenter) [reply internal/server/v1/observations/calculator.go:41]: done

**Pull request merged**